src/api/app.py

```diff
@@ -162,6 +162,15 @@ def map_date_range_to_semester_part_handler():
                 return Response(error, status=500)
     return Response("Did not receive proper form data", status=500)
 
+@app.route('/api/search', methods=['GET'])
```

Comment:
Instead of making a new route for search, let's use the `/api/class` route that can take optional search query string? so, one option could be to do `/api/class?search=abcd`. Or maybe a filters object `/api/class?filters=` and json dump the filters `{text: "abcd"}`